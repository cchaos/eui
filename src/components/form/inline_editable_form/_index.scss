@import 'inline_editable_form';
