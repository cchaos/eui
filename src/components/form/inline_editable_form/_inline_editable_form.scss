.euiInlineEditableForm {

}

.euiInlineEditableForm__inputs {
  position: relative;

  // @at-root {
  //   .euiInlineEditableForm--isEditing & {
  //     width: calc(100% - 6rem);
  //   }
  // }

  .euiFormControlLayout {
    position: relative;
    left: 0;
    width: 0;
    opacity: 0;

    @at-root {
      .euiInlineEditableForm--isEditing & {
        transition: width $euiAnimSpeedNormal $euiAnimSlightResistance $euiAnimSpeedFast;
        width: 100%;
        opacity: 1;
      }
    }
  }
}

.euiInlineEditableForm__textOnly {
  @include euiFormControlStyle;

  display: inline-block;
  margin-top: 2px;
  width: auto;

  @at-root {
    .euiInlineEditableForm--isEditing & {
      width: 0;
      opacity: 0;
    }
  }
}
