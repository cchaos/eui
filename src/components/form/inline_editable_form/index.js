export {
  EuiInlineEditableForm,
} from './inline_editable_form';
