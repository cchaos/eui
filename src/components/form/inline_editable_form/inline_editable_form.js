import React, {
  Component,
} from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { EuiButtonIcon } from '../../button';
import { EuiFieldText } from '../field_text';

export class EuiInlineEditableForm extends Component {
  static propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
  }

  constructor(props) {
    super(props);

    this.state = {
      isEditing: false,
    };
  }

  onButtonClick = () => {
    this.setState({
      isEditing: !this.state.isEditing,
    });
  }

  render() {
    const {
      //children,
      className,
      ...rest
    } = this.props;

    const classes = classNames(
      'euiInlineEditableForm',
      {
        'euiInlineEditableForm--isEditing': this.state.isEditing
      },
      className
    );

    let buttonNodes;
    if (this.state.isEditing) {
      buttonNodes = (
        <div style={{ display: 'inline' }}>
          <EuiButtonIcon iconType="minusInCircle" aria-label="cancel" onClick={this.onButtonClick} />
          <EuiButtonIcon iconType="checkInCircle" aria-label="save" onClick={this.onButtonClick} />
        </div>
      );
    } else {
      buttonNodes = (
        <EuiButtonIcon iconType="pencil" aria-label="edit" onClick={this.onButtonClick} />
      );
    }

    return (
      <div
        className={classes}
        {...rest}
      >
        <div className="euiInlineEditableForm__inputs">
          <EuiFieldText
            defaultValue="example@example.com"
            readOnly={!this.state.isEditing}
          />
          <span
            className="euiInlineEditableForm__textOnly"
            readOnly
          >
            example@example.com
          </span>

          {buttonNodes}

        </div>

      </div>
    );
  }
}
