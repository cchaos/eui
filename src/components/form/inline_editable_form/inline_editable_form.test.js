import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../../test';

import { EuiInlineEditableForm } from './inline_editable_form';

describe('EuiInlineEditableForm', () => {
  test('is rendered', () => {
    const component = render(
      <EuiInlineEditableForm {...requiredProps} />
    );

    expect(component)
      .toMatchSnapshot();
  });
});
