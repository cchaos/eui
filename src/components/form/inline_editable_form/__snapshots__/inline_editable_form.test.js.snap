// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiInlineEditableForm is rendered 1`] = `
<div
  aria-label="aria-label"
  class="euiInlineEditableForm testClass1 testClass2"
  data-test-subj="test subject string"
>
  <div
    class="euiInlineEditableForm__inputs"
  >
    <div
      class="euiFormControlLayout"
    >
      <input
        class="euiFieldText"
        readonly=""
        type="text"
        value="example@example.com"
      />
    </div>
    <span
      class="euiInlineEditableForm__textOnly"
      readonly=""
    >
      example@example.com
    </span>
    <button
      aria-label="edit"
      class="euiButtonIcon euiButtonIcon--primary"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="euiIcon euiButtonIcon__icon euiIcon--medium"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xlink="http://www.w3.org/1999/xlink"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <path
            d="M12.148 3.148L11 2l-9 9v3h3l9-9-1.144-1.144-8.002 7.998a.502.502 0 0 1-.708 0 .502.502 0 0 1 0-.708l8.002-7.998zM11 1c.256 0 .512.098.707.293l3 3a.999.999 0 0 1 0 1.414l-9 9A.997.997 0 0 1 5 15H2a1 1 0 0 1-1-1v-3c0-.265.105-.52.293-.707l9-9A.997.997 0 0 1 11 1zM5 14H2v-3l3 3z"
            id="pencil-a"
          />
        </defs>
        <use
          href="#pencil-a"
        />
      </svg>
    </button>
  </div>
</div>
`;
